src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -42,8 +42,17 @@ std::shared_ptr<SotaUptaneClient> SotaUptaneClient::newTestClient(Config &config
                                                                   std::shared_ptr<event::Channel> events_channel_in) {
   std::shared_ptr<Bootloader> bootloader_in = std::make_shared<Bootloader>(config_in.bootloader, *storage_in);
   std::shared_ptr<ReportQueue> report_queue_in = std::make_shared<ReportQueue>(config_in, http_client_in);
-  return std::make_shared<SotaUptaneClient>(config_in, storage_in, http_client_in, bootloader_in, report_queue_in,
-                                            events_channel_in);
+  auto sota_uptane_client = std::make_shared<SotaUptaneClient>(config_in, storage_in, http_client_in, bootloader_in,
+                                                               report_queue_in, events_channel_in);
+
+  // This factory method is only used for tests, so calling VirtualSecondary ctor directly here is fine
+  // for now until the virtual secondary implementation is not moved out of libaktualizr
```

Comment:
"not moved"? I think the "not" is extraneous. :)

I like this idea, but doesn't that mean that manual usage of aktualizr with virtual secondaries won't work? We still need that for various reasons. It can be temporarily solved (maybe?) by basically copying this into aktualizr-primary's main.cc.

Reply:
Yes, I added 'not' by mistake, will correct it.

>doesn't that mean that manual usage of aktualizr with virtual secondaries won't work? We still need that for various reasons. It can be temporarily solved (maybe?) by basically copying this into aktualizr-primary's main.cc.

Yeas, you're comment is reasonable, I am planning to solve it in scope of OTA-2488. The plan is to get rid of the secondary instantiation functionality out of sotauptaneclient.cc and have it out of libaktualizr so aktualizr binary executed with configuration containing secondaries, including the virtual one, will actually create and add them to the sotaclient.

Reply:
Yes, it's breaking this part of the guide https://docs.ota.here.com/quickstarts/install-a-client-locally-with-fake-secondaries.html. I would be hesitant to merge this. until there is a replacement solution

Reply:
Adding to the same factory method/iterator to SotaUptaneClient::newDefaultClient() as the one in SotaUptaneClient::newTestClient() can be a temporal solution.

Reply:
Thanks, it's working again when I've tried with a more recent version of your branch.